fix asserts now incorrect in py3

Parenthesized asserts like `assert(cond)` are incorrect in py3 - they are equiv to asserting on a tuple, and will always be True. Furthermore this is now causing a list of linter fails in the flake8 workflow.

```
./rez/release_vcs.py:[8](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:9)1:1[9](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:10): E275 missing whitespace after keyword
./rez/shells.py:429:23: E275 missing whitespace after keyword
./rez/shells.py:433:23: E275 missing whitespace after keyword
./rez/solver.py:[11](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:12)21:15: E275 missing whitespace after keyword
./rez/solver.py:[14](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:15)80:[15](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:16): E275 missing whitespace after keyword
./rez/solver.py:1535:15: E275 missing whitespace after keyword
./rez/solver.py:2205:[19](https://github.com/AcademySoftwareFoundation/rez/runs/7630360301?check_suite_focus=true#step:5:20): E275 missing whitespace after keyword
./rez/solver.py:2334:15: E275 missing whitespace after keyword
./rez/utils/scope.py:59:15: E275 missing whitespace after keyword
./rez/utils/scope.py:222:15: E275 missing whitespace after keyword
./rez/utils/scope.py:223:15: E275 missing whitespace after keyword
./rez/utils/py_dist.py:226:15: E275 missing whitespace after keyword
./rez/utils/py_dist.py:227:15: E275 missing whitespace after keyword
./rezplugins/release_vcs/git.py:64:23: E275 missing whitespace after keyword
```

--- a/src/rez/release_vcs.py
+++ b/src/rez/release_vcs.py
@@ -78,7 +78,7 @@ class ReleaseVCS(object):
                                       "path %s" % (self.name(), pkg_root))
             vcs_root = result[0]
         else:
-            assert(self.is_valid_root(vcs_root))
+            assert self.is_valid_root(vcs_root)
 
         self.vcs_root = vcs_root
         self.pkg_root = pkg_root

--- a/src/rez/shells.py
+++ b/src/rez/shells.py
@@ -426,11 +426,11 @@ class UnixShell(Shell):
             shell_command = d["command"]
         else:
             if d["stdin"]:
-                assert(self.stdin_arg)
+                assert self.stdin_arg
                 shell_command = "%s %s" % (self.executable, self.stdin_arg)
                 quiet = True
             elif do_rcfile:
-                assert(self.rcfile_arg)
+                assert self.rcfile_arg
                 shell_command = "%s %s" % (self.executable, self.rcfile_arg)
             else:
                 shell_command = self.executable

--- a/src/rez/solver.py
+++ b/src/rez/solver.py
@@ -1118,7 +1118,7 @@ class _PackageScope(_Common):
         if not package_request:
             return (self, None)
 
-        assert(new_slice is not self.variant_slice)
+        assert new_slice is not self.variant_slice
         scope = copy.copy(self)
         scope.variant_slice = new_slice
         if self.pr:
@@ -1477,7 +1477,7 @@ class _ResolvePhase(_Common):
             correctly ordered; or, if cyclic dependencies were detected, a new
             phase marked as cyclic.
         """
-        assert(self._is_solved())
+        assert self._is_solved()
         g = self._get_minimal_graph()
         scopes = dict((x.package_name, x) for x in self.scopes
                       if not x.is_conflict)
@@ -1532,7 +1532,7 @@ class _ResolvePhase(_Common):
             A 2-tuple of _ResolvePhase objects, where the first phase is the
             best contender for resolving.
         """
-        assert(self.status == SolverStatus.exhausted)
+        assert self.status == SolverStatus.exhausted
 
         scopes = []
         next_scopes = []
@@ -2202,7 +2202,7 @@ class Solver(_Common):
 
         else:
             self.pr.subheader("EXHAUSTED:")
-            assert(new_phase.status == SolverStatus.exhausted)
+            assert new_phase.status == SolverStatus.exhausted
             self._push_phase(new_phase)
 
     def failure_reason(self, failure_index=None):
@@ -2331,7 +2331,7 @@ class Solver(_Common):
         for phase in reversed(self.phase_stack):
             if phase.status not in (SolverStatus.failed, SolverStatus.cyclic):
                 return phase
-        assert(False)  # should never get here
+        assert False  # should never get here
 
     def _do_callback(self):
         keep_going = True

--- a/src/rez/utils/py_dist.py
+++ b/src/rez/utils/py_dist.py
@@ -223,8 +223,8 @@ def convert_dist(name, dest_path, make_variant=True, ignore_dirs=None,
     else:
         # this is an egg-file
         import zipfile
-        assert(is_egg and os.path.isfile(dist.location))
-        assert(zipfile.is_zipfile(dist.location))
+        assert is_egg and os.path.isfile(dist.location)
+        assert zipfile.is_zipfile(dist.location)
         z = zipfile.ZipFile(dist.location)
         z.extractall(root_path)
 

--- a/src/rez/utils/scope.py
+++ b/src/rez/utils/scope.py
@@ -56,7 +56,7 @@ class RecursiveAttribute(UserDict, StringFormatMixin):
         # to something. This stops code like "print(instance.notexist)" from
         # adding empty attributes
         attr_ = self._create_child_attribute(attr)
-        assert(isinstance(attr_, RecursiveAttribute))
+        assert isinstance(attr_, RecursiveAttribute)
         attr_.__dict__["pending"] = (attr, self)
         return attr_
 
@@ -219,8 +219,8 @@ class ScopeContext(object):
 
     def _scope_exit(self, name):
         scope = self.scope_stack.pop()
-        assert(self.scope_stack)
-        assert(name == scope.name)
+        assert self.scope_stack
+        assert name == scope.name
         data = {scope.name: scope.to_dict()}
         self.scope_stack[-1].update(data)
 

--- a/src/rezplugins/release_vcs/git.py
+++ b/src/rezplugins/release_vcs/git.py
@@ -61,7 +61,7 @@ class GitReleaseVCS(ReleaseVCS):
             try:
                 s2 = toks[-1]
                 adj, n = s2.split()
-                assert(adj in ("ahead", "behind"))
+                assert adj in ("ahead", "behind")
                 n = int(n)
                 return -n if adj == "behind" else n
             except Exception as e:

